### DISCUSSION TITLE
Oracle Driver porting or adding

### DISCUSSION THREAD

**Discussion by ElCondor1969** [mtxr/vscode-sqltools#785] [New Drivers]
Any updates or roadmap for Oracle Driver porting or adding on last version?
Thanks in advance.

**Comment by mtxr** [comment]
We need a dev for that. The maintainer the oracle driver reported to be busy.
I don't have the knowledge required on Oracle DB to get it ported, so we need someone willing to work on that to have any updates

**Comment by mtxr** [comment]
I'll convert this to a discussion since it won't be worked in this repository, but the discussion still have room to grow here

**Comment by ahsand97** [comment]
Any updates on this? I would like to contribute but  I couldn't find the old driver to get it ported so can someone point to the right direction?

**Comment by mcpiroman** [comment]
They are in previous version: https://github.com/mtxr/vscode-sqltools/tree/v0.21.9/packages/core/dialect/oracle . I have tried to port it myself at some point, but since then I went to using [dbeaver](https://github.com/dbeaver/dbeaver) OSS.

**Reply by ahsand97** [reply]
Yeah I use dbeaver too haha but it would be nice to have this extension working with Oracle too, I'm gonna try and do the porting. Thank u

**Comment by ahsand97** [comment]
Hey @mtxr @george-james-software @gjsjohnmurray, I'm trying to do the Oracle porting and I noticed that when u click the button "Test Connection" and the driver's extra dependencies are not installed (**sqlite** or **oracledb** for example) then it just fails, showing something like:

![image](https://user-images.githubusercontent.com/32344641/169194602-624566a5-4b88-4341-a9e2-a6fedf2f1478.png)

![image](https://user-images.githubusercontent.com/32344641/169194686-522768ba-644e-4891-afda-46ab3c070452.png)

So if you haven't had installed already **oracledb** (in this case) then the button "Test Connection" will always fail.

It would be nice if the dependency installation occurred similarly than with the button "Connect Now", otherwise for new people who install this extension for the first time the "Test Connection"  button is always gonna fail unless they click on "Save Connection" and then "Connect now" first so **sqltools** will ask to install the dependencies.

![image](https://user-images.githubusercontent.com/32344641/169195149-85ce80d6-da39-4945-a768-93e78d6bcf72.png)

(sqlite it's just a matter of showing the example)
![image](https://user-images.githubusercontent.com/32344641/169195172-09f73e23-9ce1-4723-8e42-91db26ffa3de.png)

I tried to make the method ` testConnection()` of the driver to ask for the installation of the dependencies but this method is not even getting executed when the dependencies aren't satisfied in the first place

```ts

public async testConnection(): Promise<void> {
    await this.checkDependencies(); // I thought this line was the solution but this method only gets executed if the dependencies are already installed
    await this.open();
   ...some logic...
    await this.close();
  }

```

**Comment by george-james-software** [comment]
Hi
It's great that you are trying to get the Oracle driver working for SQLTools.  That will be much appreciated.

What you describe sounds like a bug.  If you add it as an issue then we can investigate and see what's going on.

George

**Reply by gjsjohnmurray** [reply]
Thanks for opening https://github.com/mtxr/vscode-sqltools/issues/903

**Reply by sschneider-ihre-pvs** [reply]
is oracle now a supported driver?

**Reply by gjsjohnmurray** [reply]
@ahsand97 were you able to complete an Oracle driver extension?

**Reply by ahsand97** [reply]
> @ahsand97 were you able to complete an Oracle driver extension?

Hello there again, yes, I have some progress with the Oracle Driver and I'm planning to finally release it soon

**Reply by gjsjohnmurray** [reply]
>  I'm planning to finally release it soon

Excellent! Please let us know when this happens. We'll be happy to add it to the this of Community Drivers in the [documentation](https://vscode-sqltools.mteixeira.dev/en/home/) and README.

**Comment by bboysnick5** [comment]
Much appreciate if this can be added as vs code is my all in one solution for daily tasks.

**Comment by hashhashu** [comment]
I'm also woking on it right now and am still doing some tests and may have some issues. The project is at https://github.com/hashhashu/sqltools-Oracle-driver.

**Reply by gjsjohnmurray** [reply]
@hashhashu thanks for submitting the PR that adds your driver to the docs at https://vscode-sqltools.mteixeira.dev/en/drivers/

**Reply by MagnusMG** [reply]
Thanks a million, @hashhashu! I got it to work on OSX Ventura 13.1, VS Code 1.74.3, and Oracle Instant Client 19.3.
